packages/slate-editor/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import styled, { css } from 'styled-components';
+
+import { Text, HStack } from '#components';
+
+interface ButtonProps {
+    variant: 'clear';
+    Icon?: React.ComponentType<React.SVGProps<SVGSVGElement>>;
+    fullWidth?: boolean;
+    onClick?: () => void;
+}
+
+export function Button(props: React.PropsWithChildren<ButtonProps>) {
+    return (
+        <ButtonComponent {...props}>
+            <Text as={HStack} spacing="s" alignItems="center">
+                {props.Icon && <IconWrapper as={props.Icon} />}
+                {props.children && <span>{props.children}</span>}
+            </Text>
+        </ButtonComponent>
+    );
+}
+
+const ButtonComponent = styled.button<ButtonProps>`
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
+    outline: 0;
+    cursor: pointer;
+    text-decoration: none;
+    opacity: 0.5;
+
+    &:hover {
+        opacity: 0.8;
+    }
+
+    ${(props) =>
+        props.variant === 'clear' &&
+        css`
+            background-color: transparent;
+            border: 1px solid transparent;
+        `}
+
+    ${(props) =>
+        props.fullWidth &&
+        css`
+            flex-grow: 1;
+        `}
+`;
```

Comment:
I must say that `styled-components` is impressive in that they've built a styling framework and a whole new knowledge domain. It's clever.

What I don't like is that we cannot make it work alongside with the existing styling approach -- they have zero compatibility. We can't even reuse the same variables in boths systems.

So, with all the attractiveness, I don't think it is a good thing for the project on the current stage to switch styling to `styled-components`. I'd rather consider switching to CSS Modules, as it's more like an "evolutionary" improvement comparing to what we have now, in contrast styled components will cause a "revolution", which I don't really deal with right now.

@aspirisen Sorry about this, but I truly believe it's the right choice in the current situation. :raised_hands:

Reply:
It is possible to have the same variables, we can create a ts file that create theme values and then generate scss file with the same values. The advantage of styled-components is improved development experience. You have styles in the same file as the component, so you do not need to switch between files. As the styled components are inside TS file we can have find reference, and when the styles are no longer used TS will tell us about unused components. We would have easier refactoring when we would split components into different files we would explicitly know which styles should be also moved. Is scss we have complex classnames with concatenation and you do not have any ide support to use the class names inside react components. Classnames becomes pretty long and and the JSX starts harder to read. Also classnames becomes pretty complex when you have conditions.

In general i do not see too much difficulty in moving to styled-components as it is just css. The components that are already in scss will continue to work, and when we will change smth in the components we could change it to styled-components step by step.

By the way, plate uses styled-components as well

Reply:
> True for standard SCSS. But not the case for CSS modules.

Pretty much the same would with css modules as well, as you have element and modifiers inside the element, in jsx you have to use this long name, i.e. the element name + element--modifier + element modifier2

Reply:
I just looked through the components and it looks like there is not so many complicated scss files, the most complicated is Menu component styles, the others are simpler

Reply:
I can feel you are unhappy about reverting the styled components approach, but I still believe that it's better to not adopt it to this project at this current moment in time. 

Let me know if you could use my help on rewriting these components to SCSS.